### PR TITLE
MAKE-1004: Fix overwriting of large buffers

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: mongowire/sock.go [hunk 1]
```diff
@@ -53,13 +54,14 @@ func ReadMessage(ctx context.Context, reader io.Reader) (Message, error) {
 	if header.Size < 0 || header.Size-4 > MaxInt32 {
 		return nil, errors.New("message header has invalid size")
 	}
-	restBuf := make([]byte, header.Size-4)
+	restBuf := bytes.NewBuffer([]byte{})
```
Comment (Kimchelly) [inline mongowire/sock.go]: This can just be `&bytes.Buffer{}`.

File: mongowire/wire_message.go [hunk 2]
```diff
@@ -212,10 +216,10 @@ func (h *MessageHeader) parseMsgBody(body []byte) (Message, error) {
 			if err != nil {
 				return nil, errors.Wrap(err, "could not read identifier")
 			}
-			loc += len(section.Identifier)
+			loc += len(section.Identifier) + 1
 
-			for remaining := int(section.Size) - 4 - len(section.Identifier); remaining > 0; {
-				docSize := int(readInt32(body[loc+1:]))
+			for remaining := int(section.Size) - 1 - 4 - len(section.Identifier) - 1; remaining > 0; {
```
Comment (tychoish) [inline mongowire/wire_message.go]: it feels like the +1 on line 219 and then the -1 here could maybe be a bit clearer.

Comment (julianedwards) [reply mongowire/wire_message.go]: its for the null terminator of c strings

Comment (julianedwards) [reply mongowire/wire_message.go]: added a comment above

**Pull request merged**